x/ibc/04-channel/keeper/handshake.go

```diff
@@ -47,26 +49,19 @@ func (k Keeper) ChanOpenInit(
 		return sdkerrors.Wrap(connection.ErrConnectionNotFound, connectionHops[0])
 	}
 
-	if connectionEnd.State == connection.UNINITIALIZED {
+	if connectionEnd.GetState() == connectionexported.UNINITIALIZED {
 		return sdkerrors.Wrap(
 			connection.ErrInvalidConnectionState,
 			"connection state cannot be UNINITIALIZED",
 		)
 	}
 
-	/*
-		// TODO: Maybe not right
-		key := sdk.NewKVStoreKey(portID)
-
-		if !k.portKeeper.Authenticate(key, portID) {
-			return sdkerrors.Wrap(port.ErrInvalidPort, portID)
-		}
-
-	*/
-	channel := types.NewChannel(types.INIT, order, counterparty, connectionHops, version)
+	channel := types.NewChannel(exported.INIT, order, counterparty, connectionHops, version)
 	k.SetChannel(ctx, portID, channelID, channel)
 
 	// TODO: generate channel capability key and set it to store
```

Comment:
This is still TODO but it's blocked on ADR 3 implementation.

Reply:
Maybe leave a note here with that PR?